src/profiler/profiler.h

```diff
@@ -131,8 +131,9 @@ struct ProfileStat {
   size_t process_id_ = current_process_id();
 
   /*! \brief id of thread which operation run on */
-  std::thread::id thread_id_ = std::this_thread::get_id();  // Not yet seen a
-                                                            // case where this isn't valid
+  size_t thread_id_ = std::hash<std::thread::id>{}(std::this_thread::get_id());
```

Comment:
why do it here instead of emit time?

Reply:
Does each ProfileStat get used only once? I thought this would be better, because we can hash once only.

Reply:
only once

Reply:
it's no longer thread_id_ if you do this...